src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,85 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
+                      VerifyOrReturn(handler != nullptr);
 
-                    response.imageURI.SetValue(uri);
-                    handler->AddResponse(cachedCommandPath, response);
-                    handle.Release();
-                    return;
-                }
-                if (!hasUpdate) {
-                    // Send whatever error response our delegate decided on.
-                    handler->AddResponse(cachedCommandPath, response);
-                } else {
-                    // We must have isBDXProtocolSupported false.  Send the corresponding error status.
-                    Commands::QueryImageResponse::Type protocolNotSupportedResponse;
-                    protocolNotSupportedResponse.status
-                        = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
-                    handler->AddResponse(cachedCommandPath, protocolNotSupportedResponse);
-                }
-                handle.Release();
-                gOtaSender.ResetState();
-            }
+                      ChipLogDetail(Controller, "QueryImage: application responded with: %s",
+                          [[data description] cStringUsingEncoding:NSUTF8StringEncoding]);
 
-                          errorHandler:^(NSError *) {
-                              // Not much we can do here
-                          }];
-    };
+                      Commands::QueryImageResponse::Type delegateResponse;
+                      ConvertFromQueryImageResponseParms(data, delegateResponse);
+
+                      // If BDX protocol is not supported, send a response with status - Protocol Not supported and reset state.
+                      if (![commandParams.protocolsSupported
+                              containsObject:@(MTROtaSoftwareUpdateProviderOTADownloadProtocolBDXSynchronous)]) {
+                          Commands::QueryImageResponse::Type protocolNotSupportedresponse;
+                          protocolNotSupportedresponse.status
+                              = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                          handler->AddResponse(cachedCommandPath, protocolNotSupportedresponse);
+                          handle.Release();
+                          gOtaSender.ResetState();
+                      } else if (![data.status isEqual:@(MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable)]) {
+
+                          // If BDX protocol is supported but update is not available, send the delegate response and reset state
+                          handler->AddResponse(cachedCommandPath, delegateResponse);
+                          handle.Release();
+                          gOtaSender.ResetState();
+                      } else {
+
+                          // If there is an update available, return a response with status - Available and image uri set in the
```

Comment:
And this whole block would get outdented if we take early returns on errors above.

Reply:
:)